modules/cas_cache/layer_cache_management.c

```diff
@@ -1709,10 +1709,15 @@ int cache_mngt_set_partitions(const char *cache_name, size_t name_len,
 
 	result = ocf_mngt_cache_get_by_name(cas_ctx, cache_name,
 					name_len, &cache);
+
 	if (result)
 		goto out_get;
 
 	for (class_id = 0; class_id < OCF_USER_IO_CLASS_MAX; class_id++) {
+		
+		if (ocf_cache_is_standby(cache))
+			return -OCF_ERR_CACHE_STANDBY;
```

Comment:
The memory allocated in line 1693 needs to be freed. Also you need to decrement reference counter incremented in line 1710. Alternatively, you can check if cache is in standby mode at the very beginning of function before allocating memory. However, this would require acquiring a reference to the cache a few lines sooner than now and a small redesign of error handling.